chapters/dynamic_state.md

```diff
@@ -0,0 +1,65 @@
+# Pipeline Dynamic State
+
+> [Vulkan Spec chapter](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#pipelines-dynamic-state)
+
+## Overview
+
+When creating a graphics `VkPipeline` object the logical flow for setting state is:
+
+```cpp
+// Using viewport state as an example
+VkViewport viewport = {0.0, 0.0, 32.0, 32.0, 0.0, 1.0};
+
+// Set value of state
+VkPipelineViewportStateCreateInfo viewportStateCreateInfo;
+viewportStateCreateInfo.pViewports = &viewport;
```

Comment:
For the particular example you're giving with VK_DYNAMIC_STATE_VIEWPORT I would add this to the pseudo-code:
```c++
viewportStateCreateInfo.viewportCount = 1;
```

Later it'll be needed even for the dynamic case.